rfcs/text/0000-create-file-macho.md

```diff
@@ -0,0 +1,146 @@
+|   macho.cpu.architecture             |   keyword  |   CPU architecture target for the file.                                     |
+|   macho.cpu.byte_order               |   keyword  |   CPU byte order for the file.                                              |
+|   macho.cpu.subtype                  |   keyword  |   CPU subtype for the file.                                                 |
+|   macho.cpu.type                     |   keyword  |   CPU type for the file.                                                    |
+|   macho.headers                      |   nested   |   Header information for the file.                                          |
+|   macho.headers.commands.number      |   long     |   Number of load commands for the Mach-O header.                            |
+|   macho.headers.commands.size        |   long     |   Size of load commands of the Mach-O header.                               |
+|   macho.headers.commands.type        |   keyword  |   Type of the load commands for the Mach-O header.                          |
+|   macho.headers.magic                |   keyword  |   Magic field of the Mach-O header.                                         |
+|   macho.headers.flags                |   keyword  |   Flags set in the Mach-O header.                                           |
+|   macho.segments                     |   nested   |   Segment information for the file.                                         |
+|   macho.segments.vmaddr              |   keyword  |   Memory address of this segment.                                           |
+|   macho.segments.name                |   keyword  |   Name of this segment.                                                     |
+|   macho.segments.vmsize              |   keyword  |   Memory size of this segment.                                              |
+|   macho.segments.fileoff             |   keyword  |   File offset of this segment.                                              |
+|   macho.segments.filesize            |   keyword  |   Amount of memory to map from the file.                                    |
+|   macho.segments.sections            |   flattened   |   Section information for the segment of the file.                          |
+|   macho.segments.sections.flags      |   keyword  |   Section flags for the segment of the file.                                |
+|   macho.segments.sections.name       |   keyword  |   Section name for the segment of the file.                                 |
+|   macho.segments.sections.type       |   keyword  |   Section type for the segment of the file.                                 |
+|   macho.segments.offset      |   long  |   Offset of the segment.                                             |
+|   macho.segments.size     |   keyword  |   Segment limit size.                                            |
+|   macho.segments.flags     |   keyword  |   Segment flags.                                                 |
```

Comment:
This suggestion matches the data interface to ELF, using consistent naming (physical offset vs file offset for example), aligns segments and sections information, and adds `entropy` and  `chi2` values to `macho.sections`.
```suggestion
|   macho.segments                 |   nested   |   Segment information for the file.                   |
|   macho.segments.name            |   keyword  |   Name of this segment.                               |
|   macho.segments.physical_offset |   long     |   File offset of this segment.                        |
|   macho.segments.physical_size   |   keyword  |   Amount of memory to map from the file.              |
|   macho.segments.virtual_address |   keyword  |   Memory address of this segment.                     |
|   macho.segments.virtual_size    |   keyword  |   Memory size of this segment.                        |
|   macho.segments.sections        |   keyword  |   Section names contained in this segment.            |
|   macho.sections                 |   nested   |   Section information for the segment of the file.    |
|   macho.sections.name            |   keyword  |   Section name for the segment of the file.           |
|   macho.sections.flags           |   keyword  |   Section flags for the segment of the file.          |
|   macho.sections.type            |   keyword  |   Section type for the segment of the file.           |
|   macho.sections.physical_offset |   long     |   Section List offset.                                |
|   macho.sections.physical_size   |   long     |   Section List physical size.                         |
|   macho.sections.virtual_address |   long     |   Section List virtual address.                       |
|   macho.sections.virtual_size    |   long     |   Section List virtual size.                          |
|   macho.sections.entropy         |   float    |   Shannon entropy calculation from the section.       |
|   macho.sections.chi2            |   float    |   Chi-square probability distribution of the section. |
```